pkg/cluster/check/default.go

```diff
@@ -103,70 +203,40 @@ func ExtraClusterChecks() []ClusterCheck {
+	return []string{
+		CheckEtcdHealthy,
+		CheckEtcdConsistent,
+		CheckEtcdControlPlane,
+		CheckApidReady,
+		CheckAllNodesMemorySizes,
+		CheckAllNodesDiskSizes,
+		CheckNoDiagnostics,
+		CheckKubeletHealthy,
+		CheckAllNodesBootSequenceFinished,
+	}
+}
+
 // PreBootSequenceChecks returns a set of Talos cluster readiness checks which are run before boot sequence.
 func PreBootSequenceChecks() []ClusterCheck {
-	return []ClusterCheck{
-		// wait for etcd to be healthy on all control plane nodes
-		func(cluster ClusterInfo) conditions.Condition {
-			return conditions.PollingCondition("etcd to be healthy", func(ctx context.Context) error {
-				return ServiceHealthAssertion(ctx, cluster, "etcd", WithNodeTypes(machine.TypeInit, machine.TypeControlPlane))
-			}, 5*time.Minute, 5*time.Second)
-		},
-
-		// wait for etcd members to be consistent across nodes
-		func(cluster ClusterInfo) conditions.Condition {
-			return conditions.PollingCondition("etcd members to be consistent across nodes", func(ctx context.Context) error {
-				return EtcdConsistentAssertion(ctx, cluster)
-			}, 5*time.Minute, 5*time.Second)
-		},
-
-		// wait for etcd members to be the control plane nodes
-		func(cluster ClusterInfo) conditions.Condition {
-			return conditions.PollingCondition("etcd members to be control plane nodes", func(ctx context.Context) error {
-				return EtcdControlPlaneNodesAssertion(ctx, cluster)
-			}, 5*time.Minute, 5*time.Second)
-		},
-
-		// wait for apid to be ready on all the nodes
-		func(cluster ClusterInfo) conditions.Condition {
-			return conditions.PollingCondition("apid to be ready", func(ctx context.Context) error {
-				return ApidReadyAssertion(ctx, cluster)
-			}, 5*time.Minute, 5*time.Second)
-		},
-
-		// wait for all nodes to report their memory size
-		func(cluster ClusterInfo) conditions.Condition {
-			return conditions.PollingCondition("all nodes memory sizes", func(ctx context.Context) error {
-				return AllNodesMemorySizes(ctx, cluster)
-			}, 5*time.Minute, 5*time.Second)
-		},
-
-		// wait for all nodes to report their disk size
-		func(cluster ClusterInfo) conditions.Condition {
-			return conditions.PollingCondition("all nodes disk sizes", func(ctx context.Context) error {
-				return AllNodesDiskSizes(ctx, cluster)
-			}, 5*time.Minute, 5*time.Second)
-		},
-
-		// check diagnostics
-		func(cluster ClusterInfo) conditions.Condition {
-			return conditions.PollingCondition("no diagnostics", func(ctx context.Context) error {
-				return NoDiagnostics(ctx, cluster)
-			}, time.Minute, 5*time.Second)
-		},
-
-		// wait for kubelet to be healthy on all
-		func(cluster ClusterInfo) conditions.Condition {
-			return conditions.PollingCondition("kubelet to be healthy", func(ctx context.Context) error {
-				return ServiceHealthAssertion(ctx, cluster, "kubelet", WithNodeTypes(machine.TypeInit, machine.TypeControlPlane))
-			}, 5*time.Minute, 5*time.Second)
-		},
+	names := preBootSequenceCheckNames()
```

Comment:
this won't pass linting - `make lint-go`

and this should be `return PreBootSequenceChecksFiltered(nil)`